shared/datastore.go

```diff
@@ -25,6 +25,23 @@ func LoadTestRun(ctx context.Context, id int64) (*TestRun, error) {
 	return &testRun, nil
 }
 
+// LoadTestRunsBySHAs loads all test runs that belong to any of the given revisions (SHAs).
+func LoadTestRunsBySHAs(ctx context.Context, shas ...string) (runs TestRuns, err error) {
+	for _, sha := range shas {
+		if len(sha) > 10 {
+			sha = sha[:10]
+		}
+		var shaRuns TestRuns
+		keys, err := datastore.NewQuery("TestRun").Filter("Revision =", sha).GetAll(ctx, &shaRuns)
+		if err != nil {
+			return runs, err
```

Comment:
Should the partial list of runs be returned here? Just because this isn't nullable, or what's the convention? (Sorry still a golang noob.)

Reply:
A partial result _should_ be ignored by the caller, but sometimes it's useful to pass the results that actually loaded prior to the error.